src/main/java/org/jenkinsci/plugins/badge/StatusImage.java

```diff
@@ -41,7 +41,7 @@
     private final String length;
 
     StatusImage(String fileName) throws IOException {
-        etag = Jenkins.RESOURCE_PATH+'/'+fileName;
+        etag = '"' + Jenkins.RESOURCE_PATH + '/' + fileName + '"';
```

Comment:
Hm, that's a good question. Actually, it looks like a double quote in the string is invalid. According the comment [in the rfc 7232](http://tools.ietf.org/html/rfc7232#section-2.3) it used to be valid with backslash escaping in the obsoleted rfc 2616.

```
ETag       = entity-tag

entity-tag = [ weak ] opaque-tag
weak       = %x57.2F ; "W/", case-sensitive
opaque-tag = DQUOTE *etagc DQUOTE
etagc      = %x21 / %x23-7E / obs-text
           ; VCHAR except double quotes, plus obs-text
```

So maybe we should replace the ETag by something like sha1(fileName)?